chia/data_layer/data_store.py

```diff
@@ -350,6 +405,33 @@ async def get_ancestors(self, node_hash: bytes32, tree_id: bytes32, *, lock: boo
 
         return ancestors
 
+    async def get_ancestors_2(self, node_hash: bytes32, tree_id: bytes32, lock: bool = True) -> List[InternalNode]:
+        # TODO: implement with RECURSIVE.
+        nodes = []
+        async with self.db_wrapper.locked_transaction(lock=lock):
+            root = await self.get_tree_root(tree_id=tree_id, lock=False)
+            if root.node_hash is None:
+                raise Exception(f"Root hash is unspecified for tree ID: {tree_id.hex()}")
+            while node_hash != root.node_hash:
+                cursor = await self.db.execute(
+                    """
+                    SELECT hash, ancestor FROM ancestors
```

Comment:
I would think a recursive query like already in use would be much faster than bouncing back and forth between Python and SQLite?  I can certainly explain those queries if you want.  It took me awhile to get my head around them but after that they are ok to read..

Reply:
you're right - it's at least twice as fast as before. can you please double check the query is correct? it passes against `skip_expansive_checks` (i.e. against old `get_ancestors` method).